operator/v1/types_klusterlet.go

```diff
@@ -87,25 +90,30 @@ type KlusterletSpec struct {
 	// +optional
 	HubApiServerHostAlias *HubApiServerHostAlias `json:"hubApiServerHostAlias,omitempty"`
 
-	// ResourceRequirement specify QoS classes of klusterlet deployment
+	// ResourceRequirement specify QoS classes of deployments managed by klusterlet
 	// +optional
 	ResourceRequirement *ResourceRequirement `json:"resourceRequirement,omitempty"`
 }
 
 type ResourceQosClass string
 
 const (
-	// Default use resource setting in the template file
+	// Default use resource setting in the template file (with requests but no limits in the resources)
 	ResourceQosClassDefault ResourceQosClass = "Default"
 	// If all containers in the pod don't set resource request and limits, the pod is treated as BestEffort.
 	ResourceQosClassBestEffort ResourceQosClass = "BestEffort"
+	// Configurable resources with requests and limits
+	ResourceQosClassConfigurable ResourceQosClass = "Configurable"
 )
 
 // ResourceRequirement allow user override the default pod QoS classes
 type ResourceRequirement struct {
 	// +kubebuilder:validation:Enum=Default;BestEffort
 	// +kubebuilder:default:=Default
 	Type ResourceQosClass `json:"type"`
+	// Resources defines resource requests and limits when Type is ResourceQosClassConfigurable
+	// +optional
+	Resources *corev1.ResourceList `json:"resources,omitempty"`
```

Comment:
You would need to add ResourceQosClassConfigurable in the // +kubebuilder:validation:Enum=Default;BestEffort for type.

I would suggest the type is ResourceRequirements and the field name should be ResourceRequirements too.

Reply:
Added ResourceQosClassConfigurable in the validation comment

> I would suggest the type is ResourceRequirements and the field name should be ResourceRequirements too
Do you mean changing 
```
ResourceRequirement *ResourceRequirement `json:"resourceRequirement,omitempty"`
```
to
```
ResourceRequirements *ResourceRequirements `json:"resourceRequirement,omitempty"`
```
But `json:"resourceRequirement,omitempty"` has to stay unchanged, right?

Reply:
ResourceRequirement *ResourceRequirement `json:"resourceRequirement,omitempty"` is ok